### ISSUE TITLE
Fix Grid pagination example a11y

### ISSUE THREAD

**Issue by rolfsmeds** [vaadin/docs#5762]
The [pagination example for Grid](https://vaadin.com/docs/latest/components/grid#pagination) has a `Page X of Y` text that should be made into a live region so that screen readers announce when the page changes.

Also the Page Size label should be `aria-labelledby`-associated with its field.
